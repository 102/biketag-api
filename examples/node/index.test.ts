// @ts-ignore
import { getGameDataAsync, getTagAsync, BikeTagClient, sanityInstanceOpts, imgurInstanceOpts, getLatestTagAsync, getTagsAsync, getTag1Async } from './index'

test('BikeTagClient sanity instance can retrieve game data', async () => {
  const client = new BikeTagClient(sanityInstanceOpts)
  const testGameData = await getGameDataAsync(client)

  expect(testGameData).toMatchSnapshot()
})

test('BikeTagClient imgur instance can retrieve tag data', async () => {
  const client = new BikeTagClient(imgurInstanceOpts)
  const testTagData = await getTagAsync(client)

  expect(testTagData).toMatchSnapshot()
})

test('BikeTagClient sanity instance can retrieve tag data', async () => {
  const client = new BikeTagClient(sanityInstanceOpts)
  const testTagData = await getTagAsync(client)

  expect(testTagData).toMatchSnapshot()
})

test('BikeTagClient imgur instance can retrieve tag 1 data', async () => {
  const client = new BikeTagClient(imgurInstanceOpts)
  const testTagData = await getTag1Async(client)

  expect(testTagData).toMatchSnapshot()
})

test('BikeTagClient sanity instance can retrieve tag 1 data', async () => {
  const client = new BikeTagClient(sanityInstanceOpts)
  const testTagData = await getTag1Async(client)

  expect(testTagData).toMatchSnapshot()
})

// test('BikeTagClient reddit instance can retrieve latest tag data', async () => {
//   const client = new BikeTagClient(redditInstanceOpts)
//   const testTagData = await getTagAsync(client)

//   expect(testTagData).toMatchObject({
//     data: {
//       tagnumber: 1,
//       name: 'test-tag-1',
//       slug: 'test-tag-1',
//       game: 'test',
//       discussionUrl: undefined,
//       foundLocation: 'nowhere?',
//       player: 'hooooo',
//       hint: null,
//       mysteryImageUrl: 'https://i.imgur.com/6cXbPlZ.jpg',
//       foundImageUrl: 'https://i.imgur.com/UATbr0p.jpg',
//       gps: { lat: 0, long: 0, alt: 0 },
//     },
//     source: "imgur",
//     status: 200,
//     success: true,
//   })
// })

test('BikeTagClient imgur instance can retrieve latest tag data', async () => {
  const client = new BikeTagClient(imgurInstanceOpts)
  const testTagData = await getLatestTagAsync(client)

  expect(testTagData).toMatchSnapshot()
})

test('BikeTagClient sanity instance can retrieve latest tag data', async () => {
  const client = new BikeTagClient(sanityInstanceOpts)
  const testTagData = await getLatestTagAsync(client)

  expect(testTagData).toMatchSnapshot()
})

test('BikeTagClient imgur instance can retrieve all tags data', async () => {
  const client = new BikeTagClient(imgurInstanceOpts)
  const testTagsData = await getTagsAsync(client)

  expect(testTagsData).toMatchSnapshot()
})

test('BikeTagClient sanity instance can retrieve all tags data', async () => {
  const client = new BikeTagClient(sanityInstanceOpts)
  const testTagsData = await getTagsAsync(client)

  expect(testTagsData).toMatchSnapshot()
})