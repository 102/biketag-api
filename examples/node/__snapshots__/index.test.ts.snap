// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikeTagClient imgur instance can retrieve all tags data 1`] = `
Object {
  "data": Array [
    Object {
      "discussionUrl": undefined,
      "foundImageUrl": "https://i.imgur.com/UATbr0p.jpg",
      "foundLocation": "nowhere?",
      "game": "test",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": null,
      "mysteryImageUrl": "https://i.imgur.com/6cXbPlZ.jpg",
      "name": "test-tag-1",
      "player": "hooooo",
      "slug": "test-tag-1",
      "tagnumber": 1,
    },
    Object {
      "discussionUrl": undefined,
      "foundImageUrl": "https://i.imgur.com/FnR2O1R.jpg",
      "foundLocation": "somewhere?",
      "game": "test",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "somelocation",
      "mysteryImageUrl": "https://i.imgur.com/9b7mpwB.jpg",
      "name": "test-tag-2",
      "player": "yooooo",
      "slug": "test-tag-2",
      "tagnumber": 2,
    },
    Object {
      "discussionUrl": undefined,
      "foundImageUrl": undefined,
      "foundLocation": undefined,
      "game": "test",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "hinttime",
      "mysteryImageUrl": "https://i.imgur.com/hHIJ1AW.jpg",
      "name": "test-tag-3",
      "player": "noooo",
      "slug": "test-tag-3",
      "tagnumber": 3,
    },
  ],
  "source": "imgur",
  "status": 200,
  "success": true,
}
`;

exports[`BikeTagClient imgur instance can retrieve latest tag data 1`] = `
Object {
  "data": Object {
    "discussionUrl": undefined,
    "foundImageUrl": undefined,
    "foundLocation": undefined,
    "game": "test",
    "gps": Object {
      "alt": 0,
      "lat": 0,
      "long": 0,
    },
    "hint": "hinttime",
    "mysteryImageUrl": "https://i.imgur.com/hHIJ1AW.jpg",
    "name": "test-tag-3",
    "player": "noooo",
    "slug": "test-tag-3",
    "tagnumber": 3,
  },
  "source": "imgur",
  "status": 200,
  "success": true,
}
`;

exports[`BikeTagClient imgur instance can retrieve tag 1 data 1`] = `
Object {
  "data": Object {
    "discussionUrl": undefined,
    "foundImageUrl": "https://i.imgur.com/UATbr0p.jpg",
    "foundLocation": "nowhere?",
    "game": "test",
    "gps": Object {
      "alt": 0,
      "lat": 0,
      "long": 0,
    },
    "hint": null,
    "mysteryImageUrl": "https://i.imgur.com/6cXbPlZ.jpg",
    "name": "test-tag-1",
    "player": "hooooo",
    "slug": "test-tag-1",
    "tagnumber": 1,
  },
  "source": "imgur",
  "status": 200,
  "success": true,
}
`;

exports[`BikeTagClient imgur instance can retrieve tag data 1`] = `
Object {
  "data": Object {
    "discussionUrl": undefined,
    "foundImageUrl": "https://i.imgur.com/UATbr0p.jpg",
    "foundLocation": "nowhere?",
    "game": "test",
    "gps": Object {
      "alt": 0,
      "lat": 0,
      "long": 0,
    },
    "hint": null,
    "mysteryImageUrl": "https://i.imgur.com/6cXbPlZ.jpg",
    "name": "test-tag-1",
    "player": "hooooo",
    "slug": "test-tag-1",
    "tagnumber": 1,
  },
  "source": "imgur",
  "status": 200,
  "success": true,
}
`;

exports[`BikeTagClient reddit instance can retrieve all tags data 1`] = `
Object {
  "data": Array [
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/qoznhn",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/C0t2mMR",
      "foundLocation": "Laurelhurst Park Pond (East Side",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/jO8DY7q.jpg",
      "name": "",
      "player": "Instagram.com/y_o_p",
      "slug": "",
      "tagnumber": 515,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/qnvchj",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/VKa6wCO",
      "foundLocation": "Rossi farms on ne 122",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/lSnzr47.jpg",
      "name": "",
      "player": "M & G",
      "slug": "",
      "tagnumber": 514,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/qin3t2",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/sFIsfaT",
      "foundLocation": "st John's bridge",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/THkpkIS.jpg",
      "name": "",
      "player": "blewa",
      "slug": "",
      "tagnumber": 513,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/qez9lx",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/9LvFAe6",
      "foundLocation": "Slid more and interstate",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/DiplXN0.jpg",
      "name": "",
      "player": "Greazysteak",
      "slug": "",
      "tagnumber": 512,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/qdupz6",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/motxPOr",
      "foundLocation": "Peninsula Park",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/zJGmgOc.jpg",
      "name": "",
      "player": "I/livehorribly",
      "slug": "",
      "tagnumber": 511,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/qbn7gz",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/B31nBbD",
      "foundLocation": "alley behind Airway Science for kids on N Mississippi",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/GrrGysL.jpg",
      "name": "",
      "player": "The dynamic duo",
      "slug": "",
      "tagnumber": 510,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/q69yrt",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/BH1OIjQ",
      "foundLocation": "Broadway Bridge",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/C8I2Q2P.jpg",
      "name": "",
      "player": "u/livehorribly",
      "slug": "",
      "tagnumber": 509,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/q5j7w4",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/iS4nnGm",
      "foundLocation": "Earl bridge",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/qkdGS01.jpg",
      "name": "",
      "player": "M & G & J",
      "slug": "",
      "tagnumber": 508,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/q4tynh",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/pfKFzfm",
      "foundLocation": "NE Interstate and Oregon",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/mLGnCt2.jpg",
      "name": "",
      "player": "@tallerpauler",
      "slug": "",
      "tagnumber": 507,
    },
    Object {
      "_id": undefined,
      "_type": undefined,
      "discussionUrl": "https://redd.it/q4a800",
      "foundImage": undefined,
      "foundImageUrl": "https://imgur.com/oFu5bTm",
      "foundLocation": "SW 5th and Lincoln",
      "game": "",
      "gps": "",
      "hint": "",
      "mysteryImage": undefined,
      "mysteryImageUrl": "https://i.imgur.com/UTUeqxY.jpg",
      "name": "",
      "player": "u/livehorribly",
      "slug": "",
      "tagnumber": 506,
    },
  ],
  "source": "reddit",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient reddit instance can retrieve latest tag data 1`] = `
Object {
  "data": Object {
    "_id": undefined,
    "_type": undefined,
    "discussionUrl": "https://redd.it/qoznhn",
    "foundImage": undefined,
    "foundImageUrl": "https://imgur.com/C0t2mMR",
    "foundLocation": "Laurelhurst Park Pond (East Side",
    "game": "",
    "gps": "",
    "hint": "",
    "mysteryImage": undefined,
    "mysteryImageUrl": "https://i.imgur.com/jO8DY7q.jpg",
    "name": "",
    "player": "Instagram.com/y_o_p",
    "slug": "",
    "tagnumber": 515,
  },
  "source": "reddit",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient reddit instance can retrieve tag 1 data 1`] = `
Object {
  "data": Object {
    "_id": undefined,
    "_type": undefined,
    "discussionUrl": "https://redd.it/qoznhn",
    "foundImage": undefined,
    "foundImageUrl": "https://imgur.com/C0t2mMR",
    "foundLocation": "Laurelhurst Park Pond (East Side",
    "game": "",
    "gps": "",
    "hint": "",
    "mysteryImage": undefined,
    "mysteryImageUrl": "https://i.imgur.com/jO8DY7q.jpg",
    "name": "",
    "player": "Instagram.com/y_o_p",
    "slug": "",
    "tagnumber": 515,
  },
  "source": "reddit",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient reddit instance can retrieve tag data 1`] = `
Object {
  "data": Object {
    "_id": undefined,
    "_type": undefined,
    "discussionUrl": "https://redd.it/qoznhn",
    "foundImage": undefined,
    "foundImageUrl": "https://imgur.com/C0t2mMR",
    "foundLocation": "Laurelhurst Park Pond (East Side",
    "game": "",
    "gps": "",
    "hint": "",
    "mysteryImage": undefined,
    "mysteryImageUrl": "https://i.imgur.com/jO8DY7q.jpg",
    "name": "",
    "player": "Instagram.com/y_o_p",
    "slug": "",
    "tagnumber": 515,
  },
  "source": "reddit",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient sanity instance can retrieve all tags data 1`] = `
Object {
  "data": Array [
    Object {
      "_id": "test-tag-1",
      "_type": "tag",
      "discussionUrl": "",
      "foundImage": null,
      "foundImageUrl": "https://i.imgur.com/UATbr0p.jpg",
      "foundLocation": "nowhere?",
      "game": "Test",
      "gps": Object {
        "lat": 37.2607322,
        "lng": -121.9323076,
      },
      "hint": "",
      "mysteryImage": null,
      "mysteryImageUrl": "https://i.imgur.com/6cXbPlZ.jpg",
      "name": "test-tag-1",
      "player": "hooooo",
      "slug": "test-tag-1",
      "tagnumber": 1,
    },
    Object {
      "_id": "test-tag-2",
      "_type": "tag",
      "discussionUrl": "",
      "foundImage": null,
      "foundImageUrl": "https://i.imgur.com/FnR2O1R.jpg",
      "foundLocation": "somewhere?",
      "game": "Test",
      "gps": Object {
        "lat": 36.9840091,
        "lng": -121.9574343,
      },
      "hint": "somelocation",
      "mysteryImage": null,
      "mysteryImageUrl": "https://i.imgur.com/9b7mpwB.jpg",
      "name": "test-tag-2",
      "player": "yooooo",
      "slug": "test-tag-2",
      "tagnumber": 2,
    },
    Object {
      "_id": "test-tag-3",
      "_type": "tag",
      "discussionUrl": "",
      "foundImage": null,
      "foundImageUrl": "",
      "foundLocation": "",
      "game": "Test",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "hinttime",
      "mysteryImage": null,
      "mysteryImageUrl": "https://i.imgur.com/hHIJ1AW.jpg",
      "name": "test-tag-3",
      "player": "noooo",
      "slug": "test-tag-3",
      "tagnumber": 3,
    },
  ],
  "source": "sanity",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient sanity instance can retrieve game data 1`] = `
Object {
  "data": Object {
    "ambassadors": Array [
      "Ken",
    ],
    "boundary": Object {
      "_type": "geopoint",
      "lat": 40.6516223,
      "lng": -119.3567677,
    },
    "logo": "",
    "mainhash": "QjnikOm",
    "name": "Test",
    "queuehash": "",
    "region": "Gerlach",
    "slug": "test",
    "subreddit": "",
  },
  "source": "sanity",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient sanity instance can retrieve latest tag data 1`] = `
Object {
  "data": Object {
    "_id": "test-tag-3",
    "_type": "tag",
    "discussionUrl": "",
    "foundImage": null,
    "foundImageUrl": "",
    "foundLocation": "",
    "game": "Test",
    "gps": Object {
      "alt": 0,
      "lat": 0,
      "long": 0,
    },
    "hint": "hinttime",
    "mysteryImage": null,
    "mysteryImageUrl": "https://i.imgur.com/hHIJ1AW.jpg",
    "name": "test-tag-3",
    "player": "noooo",
    "slug": "test-tag-3",
    "tagnumber": 3,
  },
  "source": "sanity",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient sanity instance can retrieve tag 1 data 1`] = `
Object {
  "data": Object {
    "_id": "test-tag-1",
    "_type": "tag",
    "discussionUrl": "",
    "foundImage": null,
    "foundImageUrl": "https://i.imgur.com/UATbr0p.jpg",
    "foundLocation": "nowhere?",
    "game": "Test",
    "gps": Object {
      "lat": 37.2607322,
      "lng": -121.9323076,
    },
    "hint": "",
    "mysteryImage": null,
    "mysteryImageUrl": "https://i.imgur.com/6cXbPlZ.jpg",
    "name": "test-tag-1",
    "player": "hooooo",
    "slug": "test-tag-1",
    "tagnumber": 1,
  },
  "source": "sanity",
  "status": 1,
  "success": true,
}
`;

exports[`BikeTagClient sanity instance can retrieve tag data 1`] = `
Object {
  "data": Object {
    "_id": "test-tag-1",
    "_type": "tag",
    "discussionUrl": "",
    "foundImage": null,
    "foundImageUrl": "https://i.imgur.com/UATbr0p.jpg",
    "foundLocation": "nowhere?",
    "game": "Test",
    "gps": Object {
      "lat": 37.2607322,
      "lng": -121.9323076,
    },
    "hint": "",
    "mysteryImage": null,
    "mysteryImageUrl": "https://i.imgur.com/6cXbPlZ.jpg",
    "name": "test-tag-1",
    "player": "hooooo",
    "slug": "test-tag-1",
    "tagnumber": 1,
  },
  "source": "sanity",
  "status": 1,
  "success": true,
}
`;
